### PR TITLE
Homunculus S update

### PR DESCRIPTION
**Pull request by AnisotropicDefixation**
This update deals with everything Homunculus S related, some other stuff was touched in process as well.

- General
Bosses received immunity to homunculus related statuses.
Call Homunculus: Now doesn't work while under the effect of SC__GROOMY.

- Amistr
Bulwark: Now grants +30 VIT instead of DEF, this is the correct renewal bonus.

- Sera 
Needle of Paralyze: More reasonable SC_NEEDLE_OF_PARALYZE application chance.
Painkiller: Now correctly applies level 7 Endure, uses correct damage/ASPD reduction formulas and is usable on allies.
Poison Mist: Corrected damage formula. Now properly applies SC_POISON.
Summon Legion: Summoned insects are no longer looters.

- Bayeri 
Stahl Horn: Now correctly makes homunculus charge at the target. Adjusted stun chance.
Stein Wand: A more accurate Safety Wall formula. Normal SW and Elemental Shield were adjusted as well as formulas were entirely wrong.
Goldene Ferse: Corrected and added status bonuses and reductions.
Angriffs Modus: Corrected and added status bonuses and reductions.

- Dieter 
Volcanic Ash: Now gives correct skill failure chance (similar to SC_FOGWALL).
Magma Flow: Corrected damage formula. Adjusted burning chance.
Granitic Armor: Can no longer be refreshed, this is so you can't avoid the penalty.
Pyroclastic: Can no longer be refreshed, this is so you can't avoid the penalty. Added a proper Hammerfall autocast chance. Now blocks weapon swapping/unequipping. This also includes some other similar cases for other jobs.

- Eira 
Overboost: Now properly applies fixed bonuses and removes 50% of master's SP upon expiration. Can no longer be refreshed and received generic 1 second cooldown.
Light of Regeneration: Corrected the amount of HP recovered upon resurrection.
Xeno Slasher: Adjusted SC_BLOODING application chance.

- Eleanor
Properly implemented and hardcoded everything, this includes combo handling (improved it for players as well), proper support for stances, tinder breaker mechanics, new SC statuses for her skills, etc.
TO DO: Maybe someone can help me with this, check line 9812 in skill.c, this worked just fine with old msg_txt but now shows "??" instead.

Might have missed something but everything seems to be working fine.

### CODE REVIEW

**Comment by Mysteries** [comment]
Do you have a reference for the changes?

**Comment by dastgir** [comment]
Also, the build is failing , please fix all compile errors too.

**Comment by AnisotropicDefixation** [comment]
@Mysteries 

Sources are iRO Wiki, WarpPortal, etc.

@dastgir 

Fixed.

File: db/re/skill_db.txt [hunk 1]
```diff
@@ -1145,14 +1145,14 @@
 8018,9,6,4,0,0x1,0,5,1,no,0,0,0,none,0,		MH_SUMMON_LEGION,Summon Legion
 8019,5,6,1,5,0,0,5,1,no,0,0,0,weapon,0,		MH_NEEDLE_OF_PARALYZE,Needle of Paralyze
 8020,5,6,2,5,0,0,5,1,no,0,0,1,weapon,0,		MH_POISON_MIST,Poison Mist
-8021,1,6,1,0,0x1,0,5,1,no,0,0,0,none,0,		MH_PAIN_KILLER,Pain Killer
-8022,0,6,4,0,0,0x1,5,1,no,0,0,0,none,0,		MH_LIGHT_OF_REGENE,Light of Regene
-8023,0,6,4,0,0,0x1,5,1,no,0,0,0,none,0,		MH_OVERED_BOOST,Overed Boost
+8021,1,6,16,0,0x1,0,5,1,no,0,0,0,none,0,	MH_PAIN_KILLER,Pain Killer
+8022,0,6,4,0,0x1,0,5,1,no,0,0,0,none,0,		MH_LIGHT_OF_REGENE,Light of Regene
+8023,0,6,4,0,0x1,',5,1,no,0,0,0,none,0,		MH_OVERED_BOOST,Overed Boost
```
Comment (exneval) [inline db/re/skill_db.txt]: Typo here

**Comment by Inzianity** [comment]
F yes.

**Comment by dastgir** [comment]
@AnisotropicDefixation 
iROWiki is not reliable for formula's
also this PR needs serious rebase, with sensible commit name, so its easy to review + check if there's some problem in future.(name should contain what that commit contains, + reference(if any, with link))

If you need any help, just say it here, we would help you rebasing.

**Comment by AnisotropicDefixation** [comment]
Most of this info is rather old info that's been available on iRO Wiki and official iRO forums for a long time, the current implementation is basically half done and missing many things, rest are just logical tweaks since many values are already based on guesswork.

This is my first time using github, which is pretty obvious by the messy amount of commits, everything was tested and is working just fine though. I could use help organizing this like you said, as it gets hard to keep track of things when working on something so large.

**Comment by dastgir** [comment]
@malufett @michieru mind checking description first(if its correct) before we can help him rebase?

**Comment by hemagx** [comment]
@dastgir  i remember Rytech shared the official formulas ~
i guess we should follow this

**Comment by zackdreaver** [comment]
http://forums.irowiki.org/showthread.php?t=101493

**Comment by Michieru** [comment]
Here is the Rytech official Share.
http://herc.ws/board/topic/4199-why-doesnt-hercules-have-homunculus-s-skills/?p=38857

**Comment by malufett** [comment]
why is it there are addon codes/changes not related to the subject also those changes must be noted..Haru don't want PR like this..
also some of the formula are wrong..

**Comment by AnisotropicDefixation** [comment]
I'll clean it up and recommit soon.

**Pull request closed without merging**